pkg/gopass/secrets/akv.go

```diff
@@ -310,3 +330,32 @@ func (a *AKV) FromMime() bool {
 func (a *AKV) SafeStr() string {
 	return "(elided)"
 }
+
+// pwWriter is a io.Writer that will extract the first line of the input stream and
+// then write it to the password field of the provided AKV. The first line can stretch
+// multiple chunks but once the first line has been completed any writes to this
+// writer will be silently discarded.
+type pwWriter struct {
+	a       *AKV
+	buf     strings.Builder
+	written bool
+}
+
+func (p *pwWriter) Write(buf []byte) (int, error) {
+	n := len(buf)
+	if p.written || p.a == nil {
+		return n, nil
+	}
+
+	i := bytes.Index(buf, []byte("\n"))
+	if i > 0 {
+		p.buf.Write(buf[0:i])
+		p.a.password = p.buf.String()
+		p.written = true
+
+		return n, nil
```

Comment:
Shouldn't this be 
```suggestion
		return i, nil
```
Since it only writes the pw?

Reply:
See my comment above. I think the approach was not great.